### PR TITLE
Add search query

### PR DESCRIPTION
**Pull request by gabibeyer**
The search query for the climatiq API golang SDK
Still need to do a couple more data types and test them, I added the issue. Thinking of doing that in the next PR

### CODE REVIEW

File: climatiq/search.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+// parseSearchRequest takes the SearchRequest struct as input
+// and encodes the parameters into the correct format
+func parseSearchRequest(req *SearchRequest) (string, error) {
+	// dataVersion is required by the climatiq API. Read their
+	// docs for more information
+	if req.DataVersion == "" {
+		return "", fmt.Errorf("error: dataVersion must be set")
+	}
+
+	// TODO: DO we let the climatiq API handle these errors instead of us?
```
Comment (Spazzy757) [inline climatiq/search.go]: Im all for leaving this to Climatiq, this way we dont have to update it here when it changes

Comment (gabibeyer) [reply climatiq/search.go]: fantastic! I'll update it!

File: climatiq/search_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+// baseURLPath is a non-empty Client baseURL path to use during tests,
+// to ensure relative URLs are used for all endpoints. See go-github
+// issue #752. The setupMockClient() functionality has been heavily inspired by
+// the go-github library setup() function.
+// https://github.com/google/go-github/blob/master/github/github_test.go#L37
+
+const testBaseURL = "/api-v3"
```
Comment (Spazzy757) [inline climatiq/search_test.go]: I know go-github has this but I dont think we need it for mocking Climatiq. we can just cat any call to the root path and fail it with a 404 response

File: climatiq/search_test.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+func setupMockClient() (client *Client, mux *http.ServeMux, serverURL string, teardown func()) {
+	// mux is the HTTP request multiplexer used with the test server.
+	mux = http.NewServeMux()
+
+	// We want to ensure that tests catch mistakes where the endpoint URL is
+	// specified as absolute rather than relative. It only makes a difference
+	// when there's a non-empty base URL path. So, use that. See issue #752.
+	apiHandler := http.NewServeMux()
+	apiHandler.Handle(testBaseURL+"/", http.StripPrefix(testBaseURL, mux))
+	apiHandler.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
```
Comment (Spazzy757) [inline climatiq/search_test.go]: Instead of all of this maybe just respond with a 404, it is more explicit

File: climatiq/search_test.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	fileLocation := fmt.Sprintf("../testdata/%v.json", payloadType)
+	file, err := os.ReadFile(fileLocation)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return file, nil
+}
+
+func TestParseSearchRequest(t *testing.T) {
```
Comment (Spazzy757) [inline climatiq/search_test.go:47]: Theres a lot of duplicated code here, looks like a good candidate for table testing (not urgent but maybe in another PR)

**Review by Spazzy757 — CHANGES_REQUESTED** [review]
couple of small changes

File: climatiq/search_test.go (hunk 4 above)
Comment (gabibeyer) [reply climatiq/search_test.go:47]: Great idea! Created an issue for v2 https://github.com/re-cinq/go-climatiq/issues/3

File: climatiq/search_test.go (hunk 2 above)
Comment (gabibeyer) [reply climatiq/search_test.go]: Good call ! Done

**Review by Spazzy757 — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**